### PR TITLE
Prevent multiple pending seek actions

### PR DESCRIPTION
**Pull request by MrTheMake**

### CODE REVIEW

**Comment by peppy** [comment]
This potentially changes the order of queued actions, which we don't want. If we want to limit the number of queued seeks, the best way would be to remove other seeks from the queue itself.

You could probably make a `SeekAction : Action` and them match by type to achieve this.

**Comment by MrTheMake** [comment]
I've combined everything into the pendingActions queue. Howevery, I am not happy with the result. The problem was that you can't inherit from the Action class.

**Comment by Nabile-Rahmani** [comment]
@MrTheMake `Action` isn't a class, it's a delegate, and those can't be inherited.
Perhaps you could try defining SeekAction as:
```csharp
delegate void SeekAction();
```
Although I'm not sure if it's allowed to be implicitly cast to `Action`.

**Comment by peppy** [comment]
I mean another approach is to just use a List instead of a Queue.

**Comment by MrTheMake** [comment]
Can a list be used though? It wouldn't be thread-safe, would it?

**Comment by smoogipooo** [comment]
Do you have a case where this is currently broken so I could inspect? In general I am fine with the changes, but I'm not sure they're really necessary?

**Comment by MrTheMake** [comment]
Yea sure. Just drag around in the music player. When releasing the mouse, the track isn't immediately seeked to the current position as there are multiple seek actions (at least 5, didn't bother to check the exact amount). Video: https://streamable.com/hmxnn

File: osu.Framework/Audio/AudioComponent.cs [hunk 1]
```diff
@@ -32,9 +36,19 @@ public virtual void Update()
...
+            Delegate queueItem;
+            while (!IsDisposed && PendingActions.TryDequeue(out queueItem))
+            {
+                var actionItem = queueItem as Action;
+                if (actionItem != null)
+                    actionItem();
+                else
+                {
+                    var seekActionItem = queueItem as SeekAction;
+                    if (seekActionItem != null && seekActionItem == LastSeekAction)
```
Comment (peppy) [inline osu.Framework/Audio/AudioComponent.cs]: this code can't be in `AudioComponent`. it's specific to `AudioTrack`

**Comment by MrTheMake** [comment]
This should be reviewable now.

**Comment by Tom94** [comment]
Why not just use atomic increment / decrement ops via `AtomicCounter`? Also, this seems like an oddly specific solution. Might be better to just add general logic for deduping certain actions within the pending actions queue.

**Comment by MrTheMake** [comment]
Superseded by https://github.com/ppy/osu-framework/pull/894

**Pull request closed without merging**